homeassistant/components/tplink/light.py

```diff
@@ -371,6 +370,7 @@ def _async_update_attrs(self) -> None:
         effect_module = self._effect_module
         if effect_module.effect != LightEffect.LIGHT_EFFECTS_OFF:
             self._attr_effect = effect_module.effect
+            self._attr_color_mode = ColorMode.BRIGHTNESS
```

Comment:
Do we need to set this back to the correct mode once the effect is switched off? It's slightly confusing because I see there's `_determine_color_mode` in `super()._async_update_attrs` but it's not clear whether that does as much as the `TPLinkLightEntity` constructor.  Is there duplicated logic in there?

Reply:
I don't think we need to do that as it's taken care of through the `_determine_color_mode`, we just override that for this special case.

I don't see duplicate logic either, as the ctor only initializes the available modes and leaves setting the active one to the `_async_update_attrs` of the corresponding class.